populus/utils/filesystem.py

```diff
@@ -85,28 +85,24 @@ def is_exe(fpath):
 
 @to_tuple
 def recursive_find_files(base_dir, pattern):
-    for dirpath, _, filenames in os.walk(base_dir):
-        for filename in filenames:
-            if fnmatch.fnmatch(filename, pattern):
-                yield os.path.join(dirpath, filename)
-
+    """
+    Recursively traverse directory tree and find files matching pattern.
 
-@to_tuple
-def find_solidity_source_files(base_dir):
-    return (
-        os.path.relpath(source_file_path)
-        for source_file_path
-        in recursive_find_files(base_dir, "*.sol")
-    )
+    :param base_dir: Base directory to start crawling.
+    :param pattern: File pattern, to match again.
+                    Either string (single pattern match) or tuple (multiple patterns)
+    """
 
+    def match(filename, pattern):
+        if isinstance(pattern, tuple):
```

Comment:
can you use the `eth_utils.is_list_like` function here so that this supports lists, tuples, and other non-string sequence types.

Reply:
definitely, neat function ;)